libraries/bezier-rs/src/subpath/lookup.rs

```diff
@@ -7,13 +9,34 @@ impl Subpath {
 	pub fn length(&self, num_subdivisions: Option<usize>) -> f64 {
 		self.iter().fold(0., |accumulator, bezier| accumulator + bezier.length(num_subdivisions))
 	}
+
+	/// Returns the segment index and `t` value that corresponds to the closest point on the curve to the provided point.
+	/// Uses a searching algorithm akin to binary search that can be customized using the [ProjectionOptions] structure.
+	pub fn project(&self, point: DVec2, options: ProjectionOptions) -> (usize, f64) {
+		if self.is_empty() {
+			panic!("Can not project to an empty Subpath")
+		}
```

Comment:
Should we add this guard to all of our subpath functions?

Reply:
Length and intersection are the only other subpath functions, and I don't believe we need to worry about this edge case for either. Length just returns 0, and intersection correctly returns an empty list